mod-infra/src/main/scala/at/forsyte/apalache/infra/passes/PassChainExecutor.scala

```diff
@@ -35,7 +36,8 @@ class PassChainExecutor(val options: WriteablePassOptions, passes: Seq[Pass]) ex
     }
 
     passes.zipWithIndex
-      .foldLeft(Option[TlaModule with TlaModuleProperties](new TlaModule("empty", Seq()) with TlaModuleProperties)) {
+      .foldLeft[Either[TExitCode, TlaModule with TlaModuleProperties]](Right[TExitCode,
+              TlaModule with TlaModuleProperties](new TlaModule("empty", Seq()) with TlaModuleProperties)) {
```

Comment:
Just a suggestion, but maybe this could be cleaned up with some temp variables? It seems a bit hairy to me now. One possibility:

```suggestion
  val emptyModule = new TlaModule("empty", Seq()) with TlaModuleProperties)
  val init = Right[TExitCode,TlaModule with TlaModuleProperties](emptyModule)
  // ...
      .foldLeft[Either[TExitCode, TlaModule with TlaModuleProperties]](init) {
```

Reply:
Good point, this is a lot more readable!